RTech.CoffeeMachine.Api/BrewCoffee/Util/BrewStatusProvider.cs

```diff
@@ -2,5 +2,25 @@
 
 public class BrewStatusProvider : IBrewStatusProvider
 {
-    public BrewStatus GetBrewStatus() => BrewStatus.Ready;
+    private readonly IWeatherProxy _weatherProxy;
+
+    public BrewStatusProvider(IWeatherProxy weatherProxy)
+    {
+        _weatherProxy = weatherProxy;
+    }
+
+    public async Task<BrewStatus> GetBrewStatus()
+    {
+        var temp = await _weatherProxy.TryGetTemperature();
+        if (!temp.HasValue)
+        {
+            throw new InvalidOperationException("Temperature retrieval failed");
```

Comment:
- Certain companies/teams have preferences on whether to fail loudly or not in API's.
- My (subjective) opinion is that aslong as API's have a global error handler (like `ErrorController.cs` in this project), then failing loudly can be considered an alternative approach (versus returning null's all the way down back to the controller level)
- Usually I adopt the preference already established in the company, since both options have downsides. One thing I consider though with throwing exceptions, is to first ensure that the global handler actually catches & handles it properly, otherwise there will be "pollution" in analytics (e.g.: azure app insights)